addons/product/models/product_template.py

```diff
@@ -410,19 +414,8 @@ def create(self, vals_list):
         # This is needed to set given values to first variant after creation
         for template, vals in zip(templates, vals_list):
             related_vals = {}
-            if vals.get('barcode'):
-                related_vals['barcode'] = vals['barcode']
-            if vals.get('default_code'):
-                related_vals['default_code'] = vals['default_code']
-            if vals.get('standard_price'):
-                related_vals['standard_price'] = vals['standard_price']
-            if vals.get('volume'):
-                related_vals['volume'] = vals['volume']
-            if vals.get('weight'):
-                related_vals['weight'] = vals['weight']
-            # Please do forward port
-            if vals.get('packaging_ids'):
-                related_vals['packaging_ids'] = vals['packaging_ids']
+            for field_name in self._get_related_fields_variant_template() & vals.keys():
+                related_vals[field_name] = vals[field_name]
```

Comment:
```suggestion
                if vals.get(field_name):
                    related_vals[field_name] = vals[field_name]
```
Didn't notice that the previous behavior only propagated values to the template if truthy, not when the values was falsy ;).

Reply:
IMO the previous behavior is wrong since it's not possible to unset if explicitly requested... The new one take care of propagating the value if it's part vals....

Reply:
@lmignon : unset is not possible in a create context. don't you think ? 
with ``if vals.get(field_name):`` it avoid useless call to write. (if no values are provided).

Reply:
yep my bad